### PR TITLE
Lots of cookie related fixes

### PR DESCRIPTION
**Pull request by dwclark**
1) Store "always on" cookies using the correct uri now.
2) Fixed wrong order header processing in ApacheHttpBuilder.
3) Fixed OkHttpBuilder to process headers correctly when there are many
headers with the same key.
4) Lots of new cookie tests

### CODE REVIEW

**Comment by dwclark** [comment]
@cjstehno This branch is ready for merge to master and for a new release. There are lots of bug fixes here.

File: build.gradle [hunk 1]
```diff
@@ -66,7 +66,7 @@ subprojects {
     }
 
     test {
-        reports.html.enabled = false
+        reports.html.enabled = true
```
Comment (cjstehno) [inline build.gradle]: I think this wont cause problems with the overall build... I have a ticket open to make sure both sets of reports work equally well.

File: http-builder-ng-core/src/test/groovy/groovyx/net/http/tk/HttpGetTestKit.groovy [hunk 2]
```diff
@@ -182,6 +182,78 @@ abstract class HttpGetTestKit extends HttpMethodTestKit {
         ]
     }
 
+    def 'cookies are only set once'() {
```
Comment (cjstehno) [inline http-builder-ng-core/src/test/groovy/groovyx/net/http/tk/HttpGetTestKit.groovy]: Nice being able to test three implementations at once isn't it? :-)

File: http-builder-ng-okhttp/src/main/java/groovyx/net/http/OkHttpBuilder.java [hunk 3]
```diff
@@ -278,7 +279,15 @@ private OkHttpFromServer(final URI uri, final Response response) {
 
         private List<Header<?>> populateHeaders() {
             final Headers headers = response.headers();
-            return headers.names().stream().map((Function<String, Header<?>>) name -> keyValue(name, headers.get(name))).collect(toList());
+            List<Header<?>> ret = new ArrayList<>();
```
Comment (cjstehno) [inline http-builder-ng-okhttp/src/main/java/groovyx/net/http/OkHttpBuilder.java]: Ah, good catch, that would be a problem. :-/

File: http-builder-ng-core/src/test/groovy/groovyx/net/http/tk/HttpGetTestKit.groovy (hunk 2 above)
Comment (dwclark) [inline http-builder-ng-core/src/test/groovy/groovyx/net/http/tk/HttpGetTestKit.groovy]: Yes, it caught bugs in every single implementation. I'm a fan.

**Pull request merged**